### DISCUSSION TITLE
Random SDP with positive definite solutions

### DISCUSSION THREAD

**Discussion by tzheng8** [cvxpy/cvxpy#1634] [Q&A]
% This is a formal Post regrading the discussion with @rileyjmurray today at Discord.

I am working on writing an random semidefinite program with CVX.  The random SDP written in the form of example always generates infeasible/unbounded solution.(https://www.cvxpy.org/examples/basic/sdp.html ). 
Note that I have fixed the symmetric matrix issue in the post, where the Matrices C and A are set to be symmetric.
Therefore, I use an additional trace constraint $\mathrm{Tr}(X) == N$ to bound the SDP. 

Note that the feasible set of an SDP problem is the positive semidefinite cone, where the set of all positive definite is the interior of the cone, and the boundary are the PSD matrices with zero eigenvalues. And I would like to generate random SDP examples, such that the optimal solutions are positive definite matrices, i.e., in the interior of the feasible set. 
However, this is not I observe when I am using the trace constraint to ensure feasibility. There are always many zero eigenvalues in the optimal solution.

**Comment by rileyjmurray** [comment]
Thanks for starting this discussion, @tzheng8! I've accepted the Overleaf invitation you sent. I see that you've added some material to get us started there. Once we develop that document more we can summarize things on this thread. I'll try to add material to the Overleaf tomorrow! (Feb 2.)
